sdk/src/main/java/io/radar/sdk/Radar.kt

```diff
@@ -2225,6 +2246,175 @@ object Radar {
+     * @param[callback] A callback.
+     */
+    @JvmStatic
+    fun sendEvent(
+        customType: String,
+        location: Location,
+        metadata: JSONObject?,
+        callback: RadarSendEventCallback
+    ) {
+        if (!initialized) {
+            callback.onComplete(RadarStatus.ERROR_PUBLISHABLE_KEY)
+
+            return
+        }
+
+        trackOnce(location, object : RadarTrackCallback {
+            override fun onComplete(status: RadarStatus, location: Location?, events: Array<RadarEvent>?, user: RadarUser?) {
+                if (status != RadarStatus.SUCCESS || location == null) {
+                    handler.post {
+                        callback.onComplete(status)
+                    }
+
+                    return
+                }
+
+                sendEvent(customType, metadata, location, events, user, callback)
+            }
+        })
+    }
+
+    /**
+     * Sends a custom event with a manually provided location.
+     *
+     * @see [](https://radar.com/documentation/api#send-a-custom-event)
+     *
+     * @param[customType] The name of the event.
+     * @param[location] The location of the event.
+     * @param[metadata] The metadata associated with the event.
+     * @param[block] A block callback.
+     */
+    @JvmStatic
+    fun sendEvent(
+        customType: String,
+        location: Location,
+        metadata: JSONObject?,
+        block: (status: RadarStatus, location: Location?, events: Array<RadarEvent>?, user: RadarUser?) -> Unit
+    ) {
+        sendEvent(customType, location, metadata, object : RadarSendEventCallback {
+            override fun onComplete(
+                status: RadarStatus,
+                location: Location?,
+                events: Array<RadarEvent>?,
+                user: RadarUser?
+            ) {
+                block(status, location, events, user)
+            }
+        })
+    }
+
+    @JvmStatic
+    private fun sendEvent(
+        customType: String,
+        metadata: JSONObject?,
+        location: Location?,
+        events: Array<RadarEvent>?,
+        user: RadarUser?,
+        callback: RadarSendEventCallback
+    ) {
+        apiClient.sendEvent(
+            customType,
+            metadata,
+            user,
+            object : RadarApiClient.RadarSendEventApiCallback {
+                override fun onComplete(status: RadarStatus, res: JSONObject?, event: RadarEvent?) {
+                    if (status != RadarStatus.SUCCESS) {
+                        handler.post {
+                            callback.onComplete(status)
+                        }
+
+                        return
+                    }
+
+                    val finalEvents: MutableList<RadarEvent>
+
+                    if (events != null && event != null) {
+                        finalEvents = events.toMutableList()
+                        finalEvents.add(0, event)
+                    } else {
+                        finalEvents = mutableListOf(event!!)
```

Comment:
Is this `!!` safe? 🤔 

Seems like iOS handles `nil` events a little differently: https://github.com/radarlabs/radar-sdk-ios/blob/develop/RadarSDK/RadarAPIClient.m#L971-L991

Reply:
You're right--Objective-C handles a potential `nil` gracefully, but this won't.

Reply:
Objective-C is more forgiving of dealing with `null` objects, so it doesn't check for `null`, and it doesn't return a server error like the Android version does. Should it?

Reply:
I think that's fine for iOS as it'll just return an empty list, correct? either way seems like a reasonable response so I don't think we should bikeshed on it too much